jdaviz/core/template_mixin.py

```diff
@@ -944,6 +944,25 @@ def selected_subset_state(self):
                         s.label == self.selected][0]
         return subset_group.subset_state
 
+    @property
+    def selected_subset_mask(self):
+        viewer_ref = getattr(self.plugin,
+                             '_default_flux_viewer_reference_name',
+                             self.viewers[0].reference_id)
```

Comment:
It looks like this does something slightly different than intended in [these lines](https://github.com/spacetelescope/jdaviz/pull/1834/files#diff-f74c71db478d08867634b32a1b41dca8d56d2a7dfd98ab49735032e5af620b31R1001-R1006). This will return the flux viewer reference name if there is one, and otherwise return the first viewer. In my implementation, I get either the spectrum viewer or the flux viewer depending on the cube dimension. This implementation looks for the flux viewer's name first always, and if that flux viewer name is not available, it returns the first viewer's reference name. I think that would be `spectrum-2d-viewer` in Cubeviz, right?

Reply:
In other words, I'm not sure this can return a spectral subset.

Reply:
good point, its a bit verbose, but do you think the next commit should handle that as expected?

Reply:
That looks better 👍🏻